src/action/finder/hash.rs

```diff
@@ -1,40 +1,42 @@
 use crate::action::finder::request::HashActionOptions;
 use crate::fs::Entry;
-use crypto::digest::Digest;
 use log::warn;
+use md5::Md5;
 use rrg_macro::ack;
+use sha1::Sha1;
+use sha2::{Digest, Sha256};
```

Comment:
Since `Digest` is shared here with `Md5`, `Sha1` and `Sha256`, I think it makes sense to state the dependency on the [`digest`] crate explicitly and import `Digest` from there.

Also, for trait improts that are brough into the scope only for the extension methods, prefer to use the [`use Foo as _;` syntax][1].

[`digest`]: https://crates.io/crates/digest

[1]: https://doc.rust-lang.org/reference/items/use-declarations.html#underscore-imports